plugins/modules/register_cluster.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+import requests
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: register_cluster
+
+short_description: Register AssistedInstaller cluster
+
+version_added: "1.0.0"
+
+description: Register an AssistedInstaller Cluster
+
+options:
+    data:
+        description: JSON data for AssistedInstaller
+        type: dict
+        requried: true
+    NOTE: AI_API_TOKEN and AI_PULL_SECRET env variables must be set
+
+author:
+    - Daniel Kostecki (@dkosteck)
+"""
+
+EXAMPLES = r"""
+# Use argument
+- name: Register cluster
+  register_cluster
+    data: {
+        'name': 'testcluster',
+        'openshift_version': '4.16',
+    }
+"""
+
+RETURN = r"""
+register_clusters:
+    description: A dict with the AssistedInstaller results
+    type: dict
+    returned: always
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+
+def run_module():
+    module_args = dict(
+        data=dict(type="dict", required=True),
```

Comment:
This is an example on how the other inputs could be done, if you feel comfortable with it: 

```suggestion
        name=dict(type="str", required=True),
        pullsecrets=dict(type="str", required=True),
        version=dict(type="str", required=True),
```